core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/preferences/TitleBlockPreferencePage.java

```diff
@@ -0,0 +1,496 @@
+    gridData.grabExcessHorizontalSpace = true;
+    gridData.verticalAlignment = GridData.FILL;
+    gridData.horizontalAlignment = GridData.FILL;
+    gridData.widthHint = 200;
+    gridData.horizontalSpan = 1;
+    v.getTable().setLayoutData(gridData);
+
+    addMenu(v);
+
+    Listener treeListener = new Listener() {
+      @Override
+      public void handleEvent(Event event) {
+        switch (event.type) {
+        case SWT.MouseDoubleClick: {
+          Point coords = new Point(event.x, event.y);
+
+          ViewerCell cell = v.getCell(coords);
+          int index = cell.getColumnIndex();
+          List<TitleBlockCell> list = (List<TitleBlockCell>) cell.getElement();
+          String name = list.get(index).name;
+          String content = list.get(index).content;
+          TitleBlockDialog dialog = new TitleBlockDialog(getShell());
+          if (content == "content") {
+            dialog.setCurrentName("");
+            dialog.setCurrentContent("");
+          } else {
+            dialog.setCurrentName(name);
+            dialog.setCurrentContent(content);
+          }
+          dialog.create();
+          if (dialog.open() == Window.OK) {
+            list.get(index).name = dialog.getName();
+            list.get(index).content = dialog.getContent();
+            cell.setText(dialog.getName());
+
+          }
+
+        }
+        }
+
+      }
+    };
+
+    v.getTable().addListener(SWT.MouseDoubleClick, treeListener);
+  }
+
+  private void disposeColumns() {
+    TableColumn[] columns = v.getTable().getColumns();
+    for (TableColumn tc : columns) {
+      tc.dispose();
+    }
+  }
+
+  private void refreshTableColumns() {
+    disposeColumns();
+    createColumns(v, columnsNumber);
+    v.setInput(updateModel(columnsNumber, rowsNumber));
+
+  }
+
+  private void refreshTableRows() {
+    rowsNumber += 1;
+    v.setInput(updateModelRows(columnsNumber, rowsNumber));
+  }
+
+  @Override
+  public boolean performOk() {
+
+    doGetPreferenceStore().setValue("defaultTitleBlock", defaultTitleBlockFieldEditor.getBooleanValue());
+    doGetPreferenceStore().setValue("columnFieldTitleBlock", columnsNumber);
+    doGetPreferenceStore().setValue("rowFieldTitleBlock", rowsNumber);
+    String s = "";
+    for (int i = 0; i < rowsNumber; i++) {
+      for (int j = 0; j < columnsNumber; j++) {
+        if (i == rowsNumber - 1 && j == columnsNumber - 1) {
+          s = s + tccMatrix.get(i).get(j).name + "+" + tccMatrix.get(i).get(j).content;
+        } else {
+          s = s + tccMatrix.get(i).get(j).name + "+" + tccMatrix.get(i).get(j).content + "+";
+
+        }
+
+      }
+    }
+    doGetPreferenceStore().setValue("tableTitleBlock", s);
+    return super.performOk();
+  }
+
+  private List<List<TitleBlockCell>> updateModel(int nrCol, int nrRows) {
+    for (int i = 0; i < nrRows; i++) {
+      List<TitleBlockCell> tbcCell = tccMatrix.get(i);
+      tbcCell.add(new TitleBlockCell("Edit", "aql:edit"));

```

Comment:
Use constants instead of inplace strings, but put an empty string in the constants instead o edit.

Reply:
Done